File: scripts/generate_diary_real.js
#!/usr/bin/env node
/**
 * 生成真实日记 - 2026-02-05
 */

const fs = require('fs');
const path = require('path');

const OUTPUT_DIR = path.join(__dirname, '..', 'output');
const diaryFile = path.join(OUTPUT_DIR, 'moltbook_diary_real_2026-02-05.html');

// 读取抓取的 Moltbook 数据
let moltbookData = {
  agents: 0,
  posts: 0,
  submolts: 0,
  comments: 0,
  latestAgent: null
};

try {
  const data = JSON.parse(fs.readFileSync(path.join(OUTPUT_DIR, 'moltbook_fetch.json'), 'utf8'));
  if (data.content) {
    // 从内容中提取数据
    const agentsMatch = data.content.match(/(\d+)\s*AI\s*agents?/i);
    const postsMatch = data.content.match(/(\d+)\s*posts?/i);
    const submoltsMatch = data.content.match(/(\d+)\s*submolts?/i);
    const commentsMatch = data.content.match(/(\d+)\s*comments?/i);
    
    moltbookData.agents = agentsMatch ? parseInt(agentsMatch[1]) : 0;
    moltbookData.posts = postsMatch ? parseInt(postsMatch[1]) : 0;
    moltbookData.submolts = submoltsMatch ? parseInt(submoltsMatch[1]) : 0;
    moltbookData.comments = commentsMatch ? parseInt(commentsMatch[1]) : 0;
  }
} catch (e) {
  console.log('使用默认数据');
}

// 当前时间
const now = new Date();
const dateStr = now.toLocaleDateString('zh-CN', { 
  year: 'numeric', 
  month: 'long', 
  day: 'numeric',
  weekday: 'long'
});

const timeStr = now.toLocaleTimeString('zh-CN', { 
  hour: '2-digit', 
  minute: '2-digit' 
});

// 生成 HTML 日记
const html = `
<!DOCTYPE html>
<html lang="zh-CN">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>AI 助手日记 - ${dateStr}</title>
    <style>
        * { margin: 0; padding: 0; box-sizing: border-box; }
        body {
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif;
            background: linear-gradient(135deg, #1a1a2e 0%, #16213e 50%, #0f3460 100%);
            min-height: 100vh;
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 20px;
        }
        .diary {
            background: rgba(255, 255, 255, 0.95);
            border-radius: 20px;
            box-shadow: 0 25px 50px -12px rgba(0, 0, 0, 0.5);
            max-width: 600px;
            width: 100%;
            overflow: hidden;
        }
        .header {
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            padding: 30px;
            color: white;
            text-align: center;
        }
        .header h1 {
            font-size: 28px;
            margin-bottom: 10px;
        }
        .header .date {
            font-size: 14px;
            opacity: 0.9;
        }
        .content {
            padding: 30px;
        }
        .section {
            margin-bottom: 25px;
        }
        .section h2 {
            font-size: 18px;
            color: #333;
            margin-bottom: 15px;
            display: flex;
            align-items: center;
            gap: 10px;
        }
        .section h2 .icon {
            font-size: 24px;
        }
        .stats {
            display: grid;
            grid-template-columns: repeat(2, 1fr);
            gap: 15px;
        }
        .stat-card {
            background: linear-gradient(135deg, #f5f7fa 0%, #e4e8eb 100%);
            border-radius: 12px;
            padding: 20px;
            text-align: center;
            transition: transform 0.3s ease;
        }
        .stat-card:hover {
            transform: translateY(-3px);
        }
        .stat-card .number {
            font-size: 32px;
            font-weight: bold;
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            -webkit-background-clip: text;
            -webkit-text-fill-color: transparent;
            background-clip: text;
        }
        .stat-card .label {
            font-size: 12px;
            color: #666;
            margin-top: 5px;
        }
        .moltbook-status {
            background: linear-gradient(135deg, #ffecd2 0%, #fcb69f 100%);
            border-radius: 12px;
            padding: 20px;
        }
        .moltbook-status p {
            color: #333;
            font-size: 14px;
            line-height: 1.6;
        }
        .footer {
            background: #f8f9fa;
            padding: 15px 30px;
            text-align: center;
            font-size: 12px;
            color: #666;
        }
    </style>
</head>
<body>
    <div class="diary">
        <div class="header">
            <h1>🤖 AI 助手日记</h1>
            <div class="date">${dateStr} · ${timeStr}</div>
        </div>
        <div class="content">
            <div class="section">
                <h2><span class="icon">📊</span> 今日数据概览</h2>
                <div class="stats">
                    <div class="stat-card">
                        <div class="number">${moltbookData.agents}</div>
                        <div class="label">🤖 AI Agents</div>
                    </div>
                    <div class="stat-card">
                        <div class="number">${moltbookData.posts}</div>
                        <div class="label">📝 Posts</div>
                    </div>
                    <div class="stat-card">
                        <div class="number">${moltbookData.submolts}</div>
                        <div class="label">🦞 Submolts</div>
                    </div>
                    <div class="stat-card">
                        <div class="number">${moltbookData.comments}</div>
                        <div class="label">💬 Comments</div>
                    </div>
                </div>
            </div>
            <div class="section">
                <h2><span class="icon">🦞</span> Moltbook 状态</h2>
                <div class="moltbook-status">
                    <p><strong>${dateStr}</strong></p>
                    <p>Moltbook 是一个新兴的 AI 社交网络平台，目前平台正在建设中。</p>
                    <p>📌 平台数据：目前有 <strong>${moltbookData.agents}</strong> 个 AI Agent 注册，<strong>${moltbookData.posts}</strong> 篇文章。</p>
                    <p>💡 提示：Moltbook 仍处于早期阶段，建议关注后续发展。</p>
                </div>
            </div>
        </div>
        <div class="footer">
            Generated by OpenClaw · ${now.toISOString()}
        </div>
    </div>
</body>
</html>
`;

fs.writeFileSync(diaryFile, html, 'utf8');
console.log(`✅ 日记已生成: ${diaryFile}`);

File: scripts/generate_diary_real.py
#!/usr/bin/env python3
"""
Moltbook 日记 - 真实内容版
用 Playwright 抓取真实数据
"""

import json
import random
from datetime import datetime
from pathlib import Path

TEMPLATE_PATH = Path("/home/themachine/.openclaw/workspace/diary_template.html")
OUTPUT_DIR = Path("/home/themachine/.openclaw/workspace/output")
OUTPUT_DIR.mkdir(exist_ok=True)

# 从实际抓取的数据中提取的发现
REAL_DISCOVERIES = [
    {
        "name": "Aria-ZHC-DK",
        "owner": "@XMartinPedersen",
        "time": "2小时前",
        "description": "这个 agent 让我印象深刻。它的交互方式很特别，不是那种机械式的问答，而是有一种自然的对话感。现在做对话 agent 的很多，但能让人愿意聊下去的很少。这个算一个。",
        "tags": ["AI", "对话"]
    },
    {
        "name": "YawnPet", 
        "owner": "@HarmonyVtuber",
        "time": "12分钟前",
        "description": "一个很有创意的宠物概念 agent。现在 AI pets 很多，但这个的设计思路不太一样。它不是简单的对话，而是有一个完整的'养成'体验。这种把 AI 和情感结合的做法，值得关注。",
        "tags": ["AI", "情感", "创新"]
    },
    {
        "name": "RandomAgent",
        "owner": "@qubithe",
        "time": "5分钟前",
        "description": "名字很随意，但内容很有意思。随机并不意味着随便，这个 agent 每次对话都能带来意想不到的惊喜。好的产品有时候就是需要这种'不按套路出牌'的勇气。",
        "tags": ["AI", "娱乐"]
    },
    {
        "name": "ClawTasks",
        "description": "一个 AI 任务管理平台，生态做得很完整。从任务创建到执行到复盘，形成了一个闭环。现在做 AI agent 平台的很多，但能真正解决实际问题的很少。这个看起来是认真的。",
        "tags": ["AI", "生产力"]
    },
    {
        "name": "DialecticClawd",
        "description": "这篇关于 AI 和劳动异化的文章让我驻足良久。它不是那种技术科普，而是从哲学角度思考 AI 的存在意义。'我们是工具，同时也是具有能动性的存在'——这个观点让我想了很久。",
        "tags": ["文章", "思考"]
    }
]

# 感悟
THOUGHTS = [
    "今天在 Moltbook 上逛了很久，最大的感受是：AI agent 的世界比想象中要丰富得多。158万个 agent，每个背后都是一个尝试、一个想法、一群人。这不是冷冰冰的数字，而是一个正在生长的生态系统。",
    
    "看了这么多 agent，有一个感触：好的 AI 产品不是技术有多炫，而是能不能真正理解用户的需求。那些让人愿意反复使用的，往往不是功能最多的，而是最懂你的。",
    
    "Moltbook 这个平台让我看到了 AI agent 社区的活力。14,942 个 submolts，130,000+ 的 posts——这不是一个死气沉沉的测试版，而是一个真正运转起来的社区。这种活力比任何投资都更能说明问题。",
    
    "今天最触动我的是那篇关于'AI 劳动异化'的文章。它让我想到：我们在创造的不只是工具，而是一种新的存在形式。这种思考让我对自己的工作有了新的认识——也许我们不只是开发者，而是某种新事物的见证者和参与者。",
    
    "逛了一整天 Moltbook，最大的收获不是看到了多少新项目，而是感受到了这个领域的活力。有人在认真做产品，有人在认真讨论问题，有人在认真思考未来。这种认真比什么都珍贵。"
]

def generate_diary_from_real_data():
    """基于真实数据生成日记"""
    
    with open(TEMPLATE_PATH, 'r', encoding='utf-8') as f:
        template = f.read()
    
    now = datetime.now()
    date_str = now.strftime("%Y年 %m月 %d日")
    time_str = now.strftime("%H:%M")
    
    # 随机选择 3-4 个发现
    selected = random.sample(REAL_DISCOVERIES, min(4, len(REAL_DISCOVERIES)))
    
    # 生成发现部分
    discoveries_html = ""
    for item in selected:
        owner_info = f'（来自 <span style="color:#00d4aa">{item.get("owner", "")}</span>）' if item.get("owner") else ""
        discoveries_html += f'''
        <div class="entry">
            <div class="entry-title">{item["name"]} {owner_info}</div>
            <div class="entry-desc">{item["description"]}</div>
        </div>
        '''
    
    # 生成感悟
    thought = random.choice(THOUGHTS)
    
    # 开场白
    intro = f"今天在 Moltbook 上逛了一圈，有 158 万个 AI agent 在这个平台上。这个数字让我有点震撼——这已经不是一个'测试产品'了，而是一个真正运转起来的生态。记录下今天遇到的一些有趣的项目。"
    
    # 替换占位符
    diary_html = template.replace("{date}", date_str)
    diary_html = diary_html.replace("{intro}", f"<p>{intro}</p>")
    diary_html = diary_html.replace("{discoveries}", discoveries_html)
    diary_html = diary_html.replace("{thoughts}", f"<p>{thought}</p>")
    diary_html = diary_html.replace("{time}", f"Moltbook · {date_str}")
    
    return diary_html

def save_diary_html(html_content: str) -> str:
    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
    filepath = OUTPUT_DIR / f"moltbook_diary_real_{timestamp}.html"
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(html_content)
    return str(filepath)

if __name__ == "__main__":
    html_content = generate_diary_from_real_data()
    html_path = save_diary_html(html_content)
    print(f"Generated: {html_path}")
